### PR TITLE
Update eval code

### PR DESCRIPTION
**Pull request by hdefazio**
Update the evaluation code to use time windows. The ground truth is also padded as shown in [this slide](https://docs.google.com/presentation/d/1GZ_P07P4rKkwO2unvXNJAydKO772obPiOlhDN0qXE8o/edit#slide=id.gffa114f24b_0_0)

### CODE REVIEW

**Comment by hdefazio** [comment]
@Purg PTAL, please forgive the git history :)

File: angel_system/eval/visualization.py [hunk 1]
```diff
@@ -1,153 +1,310 @@
...
+
+        cm = confusion_matrix(true_idxs, pred_idxs, labels=range(len(self.labels)))
+        disp = ConfusionMatrixDisplay(confusion_matrix=cm,
+                                      display_labels=self.labels)
+        disp.plot(ax=ax, xticks_rotation=90)
+        
+        plt.tight_layout()
+        fig.savefig(f"{self.output_dir}/confusion_mat.png", pad_inches=5)
+        plt.close(fig)
+
```
Comment (Purg) [inline angel_system/eval/visualization.py:234]: two new-lines separate top-level definitions, [see PEP8 section](https://peps.python.org/pep-0008/#blank-lines)

File: angel_system/eval/visualization.py [hunk 2]
```diff
@@ -1,153 +1,310 @@
...
+            # Add legend and f1 curves to plot
+            handles, labels = ax.get_legend_handles_labels()
+            handles.extend([l])
+            labels.extend(["iso-f1 curves"])
+            ax.legend(handles=handles, labels=labels, loc="best")
 
+            ax.set_xlabel("Recall")
+            ax.set_ylabel("Precision")
+
+            fig.savefig(f"{pr_plot_dir}/{label.replace(' ', '_')}.png")
```
[the remark this answers is not in the document]
Comment (hdefazio) [reply angel_system/eval/visualization.py:131]: Like a list of all filenames? I do log `log.info(f"Saved plots to {output_dir}/plots/")` in pt_evaluate.py with the hope that will point someone in the right direction to view the plots.

File: angel_system/eval/ptg_evaluate.py [hunk 3]
```diff
@@ -59,67 +62,166 @@ def run_eval(args):
...
-        type=str,
-        default="model_files/activity_detector_annotation_labels.xlsx",
-        help="Multi-sheet Excel file of class ids and labels where each sheet is titled after the label version. \
-            The sheet used for evaluation is specified by the version number in the ground truth filename"
-    )
     parser.add_argument(
         "--activity_gt",
         type=str,
-        default="data/ros_bags/labels_test_v1.2.feather",
+        default="data/ros_bags/Annotated_folding_filter/labels_test_v1.2.feather",
```
[the remark this answers is not in the document]
Comment (Purg) [reply angel_system/eval/ptg_evaluate.py]: What I meant is that I don't think this parameter should have any default: the user should be forced to provide a value here. A filename example would be better fitting to include in the help string.

File: angel_system/eval/visualization.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Purg) [reply angel_system/eval/visualization.py:115]: Right, we're in a for-loop here. I'm not sure why I originally suggested this anyway, resolving.

**Pull request merged**